File: service/src/main/java/com/jb/udp/Enum/RecordEvent.java
package com.jb.udp.Enum;

import java.util.ArrayList;
import java.util.EnumSet;

public enum RecordEvent {
    sensor((byte)(0x1 << 6)),
    phase((byte)(0x1 << 5)),
    over((byte)(0x1 << 4)),
    powerOutage((byte)(0x1 << 3)),
    dryOver((byte)(0x1 << 2)),
    wetOver((byte)(0x1 << 1));

    private byte rawValue;

    RecordEvent(byte rawValue) {
        this.rawValue = rawValue;
    }

    public static EnumSet<RecordEvent> get(byte rawValue) {
        EnumSet<RecordEvent> enumSet = EnumSet.noneOf(RecordEvent.class);

        for (RecordEvent status : RecordEvent.values()) {
            if ((rawValue & status.rawValue) == status.rawValue) {
                enumSet.add(status);
            }
        }
        return enumSet;
    }

    public static ArrayList<String> des(EnumSet<RecordEvent> enumSet) {
        ArrayList<String> arr = new ArrayList<>();

        if (enumSet.contains(sensor)) {
            arr.add("传感器故障");
        }

        if (enumSet.contains(phase)) {
            arr.add("缺相");
        }

        if (enumSet.contains(over)) {
            arr.add("过载");
        }

        if (enumSet.contains(powerOutage)) {
            arr.add("停电");
        }

        if (enumSet.contains(dryOver)) {
            arr.add("温度超限");
        }

        if (enumSet.contains(wetOver)) {
            arr.add("湿度超限");
        }

        return arr;
    }
}

File: direct/platforms/ios/WeexDemo/UDP/RecordEvent.swift
//
//  RecordEvent.swift
//  WeexDemo
//
//  Created by zzzl on 2019/1/25.
//  Copyright © 2019 zzzl. All rights reserved.
//

import Foundation

struct RecordEvent: OptionSet {
    var rawValue: UInt8
    
    static let sensor = RecordEvent(rawValue: 0x1 << 6)
    static let phase = RecordEvent(rawValue: 0x1 << 5)
    static let over = RecordEvent(rawValue: 0x1 << 4)
    static let powerOutage = RecordEvent(rawValue: 0x1 << 3)
    static let dryOver = RecordEvent(rawValue: 0x1 << 2)
    static let wetOver = RecordEvent(rawValue: 0x1 << 1)
}

extension RecordEvent {
    func des() -> [String] {
        var arr = [String]()
        
        if contains(.sensor) {
            arr.append("传感器故障")
        }
        
        if contains(.phase) {
            arr.append("缺相")
        }
        if contains(.over) {
            arr.append("过载")
        }
        if contains(.powerOutage) {
            arr.append("停电")
        }
        if contains(.dryOver) {
            arr.append("温度超限")
        }
        if contains(.wetOver) {
            arr.append("湿度超限")
        }
        return arr
    }
}
